src/AzureAuth/Commands/Ado/CommandPat.cs

```diff
@@ -89,6 +90,20 @@ private enum OutputMode
         [Option(CommandAad.TimeoutOption, CommandAad.TimeoutHelpText, CommandOptionType.SingleValue)]
         private double Timeout { get; set; } = CommandAad.GlobalTimeout.TotalMinutes;
 
+        // Scopes are normalized from application start to prevent reparsing.
+        private ImmutableSortedSet<string> _scopes;
+        private ImmutableSortedSet<string> Scopes

```

Comment:
Would it be fine to directly assign like

`private ImmutableSortedSet<string> Scopes => AdoPat.Scopes.Normalize(RawScopes);`

and not require _scopes?